pkg/maven/maven.go

```diff
@@ -103,6 +127,130 @@ func Evaluate(pomFile, expression string, command mavenExecRunner) (string, erro
 	return value, nil
 }
 
+// InstallFile installs a maven artifact and its pom into the local maven repository.
+// If "file" is empty, only the pom is installed.
+// "pomFile" may not be empty.
+func InstallFile(file, pomFile string, command mavenExecRunner, utils mavenUtils, projectSettingsFile, globalSettingsFile, m2Path string) error {
+	if len(pomFile) == 0 {
+		return fmt.Errorf("pomFile can't be empty")
+	}
+
+	var defines []string
+	if len(file) > 0 {
+		defines = append(defines, "-Dfile="+file)
+	} else {
+		defines = append(defines, "-Dfile="+pomFile)
+	}
```

Comment:
The documentation could be interpreted such that the pom file is always installed. But it is an alternative?

Reply:
I don't understand. Yes the pom file is always installed. Is there an issue with that?

Reply:
Line 142 is not executed if a file is given to be installed. So it is either-or, unless I am missing something not visible in the code.